azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json

```diff
@@ -0,0 +1,291 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Configuration",
+  "description": "Configuration for Maven plugin for Azure Web App",
+  "properties": {
+    "subscriptionId": {
+      "$ref": "#/definitions/non-empty-string"
+    },
+    "resourceGroup": {
+      "$ref": "#/definitions/azure-resource-group"
+    },
+    "appName": {
+      "type": "string",
+      "pattern": "^[a-zA-Z0-9\\-]{2,60}$"
+    },
+    "appServicePlanName": {
+      "type": "string",
+      "pattern": "^[a-zA-Z0-9\\-]{1,40}$"
+    },
+    "appServicePlanResourceGroup": {
+      "$ref": "#/definitions/azure-resource-group"
+    },
+    "auth": {
+      "$ref": "#/definitions/auth"
+    },
+    "region": {
+      "$ref": "#/definitions/non-empty-string"
+    },
+    "pricingTier": {
+      "$ref": "#/definitions/non-empty-string"
+    },
+    "runtime": {
+      "$ref": "#/definitions/runtime"
+    },
+    "deployment": {
+      "$ref": "#/definitions/deployment"
+    },
+    "deploymentSlot": {
+      "$ref": "#/definitions/deployment-slot"
+    },
+    "appSettings": {
+      "type": "object"
+    },
+    "allowTelemetry": {
+      "type": "boolean",
+      "default": true
+    },
+    "failsOnError": {
+      "type": "boolean",
+      "default": true
+    },
+    "stopAppDuringDeployment": {
+      "type": "boolean",
+      "default": false
+    },
+    "skip": {
+      "type": "boolean",
+      "default": false
+    },
+    "authType": {
+      "$ref": "#/definitions/auth-type",
+      "deprecationMessage": "Please set auth related properties like type in <auth>"
+    }
+  },
+  "required": [
+    "appName",
+    "resourceGroup"
+  ],
+  "definitions": {
+    "non-empty-string": {
+      "type": "string",
+      "minLength": 1
+    },
+    "auth-type": {
+      "type": "string",
+      "pattern": "(?i)^(auto|service_principal|managed_identity|azure_cli|vscode|intellij|azure_auth_maven_plugin|device_code|oauth2|visual_studio)$"
```

Comment:
enum?

Reply:
enum in gradle schema is case sensitive, refers https://stackoverflow.com/questions/26604730/can-json-schema-enums-be-case-insensitive